src/panoptes/utils/images/__init__.py

```diff
@@ -221,31 +222,59 @@ def _make_pretty_from_cr2(fname, title=None, **kwargs):
 
 
-def mask_saturated(data, saturation_level=None, threshold=0.9, dtype=np.float64):
-    """Convert data to masked array of requested dtype with saturated values masked.
+def mask_saturated(data, saturation_level=None, threshold=0.9, bit_depth=None, dtype=None):
+    """Convert data to a masked array with saturated values masked.
 
     Args:
         data (array_like): The numpy data array.
-        saturation_level (float, optional): The saturation level. If None,
-            the level will be set to the threshold times the max value for the dtype.
-        threshold (float, optional): The percentage of the max value to use.
-        dtype (numpy.dtype, optional): The requested dtype for the new array.
+        saturation_level (scalar, optional): The saturation level. If not given then the
+            saturation level will be set to threshold times the maximum pixel value.
+        threshold (float, optional): The fraction of the maximum pixel value to use as
+            the saturation level, default 0.9.
+        bit_depth (astropy.units.Quantity or int, optional): The effective bit depth of the
+            data. If given the maximum pixel value will be assumed to be 2**bit_depth,
+            otherwise an attempt will be made to infer the maximum pixel value from the
+            data type of the data. If data is not an integer type the maximum pixel value
+            cannot be inferred and an IllegalValue exception will be raised.
+        dtype (numpy.dtype, optional): The requested dtype for the masked array. If not given
+            the dtype of the masked array will be same as data.
 
     Returns:
         numpy.ma.array: The masked numpy array.
+
+    Raises:
+        error.IllegalValue: Raised if bit_depth is an astropy.units.Quantity object but the
+            units are not compatible with either bits or bits/pixel.
+        error.IllegalValue: Raised if neither saturation level or bit_depth are given, and data
+            has a non integer data type.
     """
     if not saturation_level:
-        try:
-            # If data is an integer type use iinfo to compute machine limits.
-            dtype_info = np.iinfo(data.dtype)
-        except ValueError:
-            # Not an integer type. Assume for now we have 16 bit data.
-            saturation_level = threshold * (2 ** 16 - 1)
+        if bit_depth is not None:
+            try:
+                with suppress(AttributeError):
+                    bit_depth = bit_depth.to_value(unit=u.bit)
+            except u.UnitConversionError:
+                try:
+                    bit_depth = bit_depth.to_value(unit=u.bit / u.pixel)
+                except u.UnitConversionError:
+                    raise error.IllegalValue("bit_depth must have units of bits or bits/pixel, " +
+                                             f"got {bit_depth!r}")
+
+            bit_depth = int(bit_depth)
+            logger.trace(f"Using bit depth {bit_depth!r}")
```

Comment:
You don't have to accept this now, but also a reminder of Python 3.8 FTW! :) 

```suggestion
            logger.trace(f"Using {bit_depth=!r}")
```